### PR TITLE
34/admin logic

### LINKED ISSUE #34: Admin logic
- Store `admin` flag on JWT & firebase.
- Use it to allow access to the Challenge review page (+ backend middleware to protect the challenge-review-submission)
- Show `admin` badge in the UI.

Dependent on #31 #32

### PR DESCRIPTION
**Pull request by carletex**
Fixes #34 

To be an admin, just add a `isAdmin = true` (boolean value) property into your user's document in Firebase.

In this PR:

- Show an "admin" badge next to your address
- Remove link to challenge review for non-admins
- Create `jwtAdminAuth` middleware
- Protect all the routes with the corresponding auth middleware. Besides the sign route, I left out the builders & builders/:address routes (But maybe we want to protect them as well)
- Tweak all the axios calls to include the headers.
-  Create an admin auth test in `/jwt-test`
- Update the `set` call on database saves to `update`. `set` overrides the whole document.

### CODE REVIEW

File: packages/backend/middlewares/auth.js [hunk 1]
```diff
@@ -54,11 +54,33 @@ const jwtAuth = (req, res, next) => {
...
+  jwtAuth(req, res, () => {
+    // Added by jwtAuth
+    if (!req.isAdmin) {
+      console.log("returning 401, Not an admin");
+      res.sendStatus(401);
+      return;
+    }
+
+    next();
+  });
```
Comment (carletex) [inline packages/backend/middlewares/auth.js:79]: @dgrcode Added the logic inside the "next" function to be sure `jwtAuth` was successful. 

If not, if `jwtAuth` throws a 401 and `jwtAdminAuth` consecutively throws another 401:
`Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client`

Comment (dgrcode) [reply packages/backend/middlewares/auth.js:79]: yep! I wasn't sure about how to chain middlewares, but using `next` looks like the best solution to me.

`jwtAuth` should only reject if the token itself is invalid. The admin rejection should happen in `jwtAdminAuth`.

File: packages/backend/index.js [hunk 2]
```diff
@@ -181,6 +178,10 @@ app.get("/auth-jwt-restricted", jwtAuth, (req, res) => {
   res.send(`all working! 👌. Successfully authenticated request from ${req.address}`);
 });
 
+app.get("/auth-jwt-admin-restricted", jwtAdminAuth, (req, res) => {
```
Comment (dgrcode) [inline packages/backend/index.js:181]: What do you think about changing the middleware names to:
 - `jwtAuth` -> `authOnly`
 - `jwtAdminAuth` -> `adminOnly`

Or something else more descriptive than the current names.

**Review by dgrcode — APPROVED** [review]

File: packages/backend/index.js (hunk 2 above)
Comment (carletex) [reply packages/backend/index.js]: I agree!

I've just changed them to `userOnly` and `adminOnly`. https://github.com/moonshotcollective/scaffold-directory/pull/38/commits/6d996b04a91a81fdee6717a4b116b42b4cbf715e

BTW, when I was testing it I got a `returning 401, invalid token. Error: jwt expired` from a jwt generated earlier today. We might want to (tackle #29 || at least increase the TTL (like 1 week/month XD) && (show a message telling the user that he has been logged out and need to re-sign)

**Pull request merged**